src/MySTMarkdownCell.tsx

```diff
@@ -34,13 +51,34 @@ export class MySTMarkdownCell
       const node = document.createElement('div');
       this.myst = { node };
     }
+
+    this._doneRendering = new PromiseDelegate<void>();
     const notebook = this.parent as StaticNotebook;
     this.myst.pre = undefined;
-    parseContent(notebook);
+    const parseComplete = parseContent(notebook);
     const widget = new Widget({ node: this.myst.node });
     widget.addClass('myst');
     this.addClass('jp-MySTMarkdownCell');
     this.inputArea.renderInput(widget);
+    if (parseComplete) {
+      parseComplete.then(() => this._doneRendering.resolve());
+    } else {
+      // Something went wrong, but we will still signal the next step?
+      this._doneRendering.resolve();
+    }
+  }
+
+  /**
+   * Whether the Markdown renderer has finished rendering.
+   */
+  get doneRendering(): Promise<void> {
+    return this._doneRendering.promise;
+  }
+
+  get expressions(): string[] {
+    const { post: mdast } = this.myst ?? {};
+    const expressions = selectAll('inlineExpression', mdast);
+    return expressions.map(node => (node as any).value);
```

Comment:
Gather all of the expressions to be evaluated. This is just the body of the role.

Reply:
So much nicer.

Reply:
Assuming that we're referring to `inlineExpression`, I'd probably prefer e.g. `expr` or `expression`?

Reply:
I guess there is some precedence in the spec at the moment.

https://myst-tools.org/docs/spec/commonmark#inline-code

I am inclined to leave it as is in this PR, and revisit when we start to standardize this in the spec?! My thinking is that we may want a difference in inline vs block-level expressions. e.g. an interactive figure inline maybe has a different renderer than the block level renderer. That could mean different mimebundles in the future.

These are all ephemeral at the moment, so it shouldn't matter if we want to change it!